### DISCUSSION TITLE
Can't use a UI on port 7999

### DISCUSSION THREAD

**Discussion by xmontero** [dunglas/mercure#784] [Q&A]
Hi! I'm absolutely new to mercure so maybe the question will seem silly for some of you.

I want to try mercure for the first time. I want to try it on docker and on a published port different from 80.

After googling [php mercure] I found here https://www.itdo.com/blog/php-y-symfony-una-breve-introduccion-a-mercure/ an introduction.

After trying multiple tests, the docker command ends up giving me this:

![image](https://github.com/dunglas/mercure/assets/5641621/bab6182e-5599-44be-b085-3d33736b7d4d)

I then googled for [mercure docker Error: loading initial config: loading new config] and I found this github issue: https://github.com/api-platform/api-platform/issues/1757 but when this comment https://github.com/api-platform/api-platform/issues/1757#issuecomment-759462771 references a docker-compose yaml for other project I'm unable to understand what it refers to.

If I enter in the docker page here https://hub.docker.com/r/dunglas/mercure It does not show any examples, and only refers to the official page here https://mercure.rocks/ and if I click on documentation and search for [docker run] like this

![image](https://github.com/dunglas/mercure/assets/5641621/eec8744f-3b81-4b26-9e9e-ee7cfcfb2848)

I end up here: https://mercure.rocks/docs/hub/install#docker-image

Then. All examples I see here assume ports 80 and 443 being published.

Here I see that what it has mainly changed from the outdated tutorial I first found is that now you pass-in two JWT instead of only one. Okey... Let's try it: I'll take any of the official docker run examples and I'll fine tune them.

For example, I take this one:

    docker run \
        -e MERCURE_PUBLISHER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -e MERCURE_SUBSCRIBER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -p 80:80 \
        -p 443:443 \
        dunglas/mercure caddy run --config /etc/caddy/Caddyfile.dev

and I remove 443 because I'm happy to use http without s for my first "hello world", and I'll change local 80 to 7999.

    docker run \
        -e MERCURE_PUBLISHER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -e MERCURE_SUBSCRIBER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -p 7999:80 \
        dunglas/mercure caddy run --config /etc/caddy/Caddyfile.dev

Okey it now seems the server does not complain. Time to test it. I've seen it seems to come with a UI for testing, doesn't it?

The documentation suggests:

    https://localhost/.well-known/mercure/ui/

I then modify it to suit my needs: Drop "s" and add port:

    http://localhost:7999/.well-known/mercure/ui/

But unfortunately the browser does not connect: For some reason loading this redirects me to `https://localhost/.well-known/mercure/ui/`

So, question:

What is the **simplest** `docker run` command to run the **simplest** server (ie: no SSL, anonymous, etc.) to make the **simplest** helloworld test using port 7999 and using the test ui?

**Comment by dunglas** [comment]
Can you try this?

```
docker run \
    -e SERVER_NAME=:7999 \
    -e MERCURE_PUBLISHER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
    -e MERCURE_SUBSCRIBER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
    -p 7999:7999 \
    dunglas/mercure caddy run --config /etc/caddy/Caddyfile.dev
```

The important part is the `SERVER_NAME` environment variable which allows to set the IP or domain and port to use: https://mercure.rocks/docs/hub/config

**Reply by xmontero** [reply]
Indeed! It works!

Thousand thanks and congrats for your great work❤️❤️❤️!! I feel this project is a game-changer and plan to use a lot! 

PD: If you can, add the missing backslash after the server name line, I can't edit yours!

**Reply by xmontero** [reply]
I'm ready to contribute. Should I change the docs so it reflects this use-case and do a PR?

**Reply by xmontero** [reply]
In fact, to be more clear about there's no need to "match the port" I've tried and it also works:

    docker run \
        -e SERVER_NAME=:80 \
        -e MERCURE_PUBLISHER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -e MERCURE_SUBSCRIBER_JWT_KEY='!ChangeThisMercureHubJWTSecretKey!' \
        -p 7999:80 \
        dunglas/mercure caddy run --config /etc/caddy/Caddyfile.dev

This example makes more explicit that the server name does not "need" to match the external port, but it's existance "disables automatic TSL switching".

Am I right?
